components/compliance-service/ingest/pipeline/processor/control_worker.go

```diff
@@ -0,0 +1,173 @@
+)
+
+var (
+	ReportWorkflowName = cereal.NewWorkflowName("control-workflow")
+	ReportTaskName     = cereal.NewTaskName("control-task")
+)
+
+const (
+	RunningStatus string = "running"
+	FailedStatus  string = "failed"
+	SuccessStatus string = "success"
+)
+
+func InitCerealManager(cerealManager *cereal.Manager, workerCount int, client *ingestic.ESClient) error {
+	logrus.Info("Successfully starting control-workflow")
+	err := cerealManager.RegisterWorkflowExecutor(ReportWorkflowName, &ControlWorkflow{})
+	if err != nil {
+		logrus.Info("Found error control-workflow")
+		logrus.Info(err)
+		return err
+	}
+
+	logrus.Info("Successfully registered control-workflow")
+	return cerealManager.RegisterTaskExecutor(ReportTaskName, &GenerateControlTask{
+		ESClient: client,
+	}, cereal.TaskExecutorOpts{Workers: workerCount})
+
+}
+
+type ControlWorkflow struct {
+}
+
+type ControlWorkflowParameters struct {
+	ReportUuid string
+	Retries    int
+}
+
+type ControlWorkflowPayload struct {
+	ReportUuid  string
+	RetriesLeft int
+	Status      string
+	StartTime   *time.Time
+}
+
+func (s *ControlWorkflow) OnStart(w cereal.WorkflowInstance,
+	ev cereal.StartEvent) cereal.Decision {
+
+	logrus.Info("In control-workflow start method")
+
+	startTime := time.Now()
+	workflowPayload := ControlWorkflowPayload{
+		StartTime: &startTime,
+	}
+
+	workflowParams := ControlWorkflowParameters{}
+	err := w.GetParameters(&workflowParams)
+	if err != nil {
+		err = errors.Wrap(err, "failed to unmarshal control-workflow parameters")
+		logrus.WithError(err).Error()
+		return w.Fail(err)
+	}
+
+	logrus.Infof("In On Start Method %s", workflowParams.ReportUuid)
+
+	workflowPayload.ReportUuid = workflowParams.ReportUuid
+	workflowPayload.RetriesLeft = workflowParams.Retries
+	workflowPayload.Status = RunningStatus
+
+	err = w.EnqueueTask(ReportTaskName, GenerateControlParameters{
+		ReportUuid: workflowParams.ReportUuid,
+	})
+	if err != nil {
+		err = errors.Wrap(err, "failed to enqueue the control-task")
+		logrus.WithError(err).Error()
+		return w.Fail(err)
+	}
+	return w.Continue(&workflowPayload)
+}
+
+func (s *ControlWorkflow) OnTaskComplete(w cereal.WorkflowInstance,
+	ev cereal.TaskCompleteEvent) cereal.Decision {
+
+	var payload ControlWorkflowPayload
+
+	if err := w.GetPayload(&payload); err != nil {
+		err = errors.Wrap(err, "failed to unmarshal control-workflow payload")
+		logrus.WithError(err).Error()
+		return w.Fail(err)
+	}
+
+	logrus.Infof("Entered ControlWorkflow > OnTaskComplete with payload %+v", payload)
+
+	if err := ev.Result.Err(); err != nil {
+		//received error, if the retries are available enqueue the task
+		if payload.RetriesLeft > 0 {
+			logrus.Debugf("retring control-task %s", payload.ReportUuid)
+
+			workflowParams := ControlWorkflowParameters{}
+			err := w.GetParameters(&workflowParams)
+			if err != nil {
+				err = errors.Wrap(err, "failed to unmarshal control-workflow parameters")
+				logrus.WithError(err).Error()
+				return w.Fail(err)
```

Comment:
We will keep on retrying as we have not decremented the reties count and returned